docs(cli): add docs for starport tools completions cmd (#1421)
tendermint/starport: docs/cli/index.md

* [starport tools](#starport-tools)	 - Tools for advanced users

## starport tools completions

Generates a completion script. 

**Synopsis**

> The ``completions`` command generates a completion script to be used in your shell. However, the generated script depends on
> [bash-completion](https://github.com/scop/bash-completion) being installed and enabled in your system. Most of *nix like
> operating systems come with it by default and we assume it is already installed and operational. 

Reviewer: ```suggestion
The ``completions`` command generates a completion script you can use in your shell. Using the `completions` command to generate a completion script requires that [bash-completion](https://github.com/scop/bash-completion) is installed and enabled in your system. Since most Unix-like operating systems come with bash-completion by default, bash-completion is probably already installed and operational. 
```

Reviewer: @andraderaf and @fadeev 
is there a way I can limit line length when I do a multi-line suggestion like this one? 
I know we are trying to limit line length? (or what is it that I can do differently here to make a PR commit of this suggestion clean?)

Reviewer 2: I don't think we should be limiting line length, at least it MD files (CLI might be another story). Users' text editors are capable of wrapping lines.

Author: I think saying "Using the `completions` command to generate a completion script requires that [bash-completion](https://github.com/scop/bash-completion) is installed and enabled in your system..." is technically inaccurate, since the command starport tool completions will work whether or not bash_completions is installed. When running the command, it outputs a functional script. That resulting script is the one that depends on bash_completions to work.

Reviewer: > I don't think we should be limiting line length, at least it MD files (CLI might be another story). Users' text editors are capable of wrapping lines.

oh thanks @fadeev I am glad my feedback won't interfere then. I recall a Slack comment you made (a few weeks ago, I think?) about using single lines in PRs. What single lines should I be aware of?

```bash
starport tools completions [bash|zsh|fish|powershell]
```

**Examples**

```
starport tools completions bash
```

**Options**

```
-h, --help   help for completions
```

### Enable starport autocompletion

> In order to enable the autocompletion every time you open a terminal, we present two solutions :

Reviewer: ```suggestion
Use one of these methods to enable the autocompletion each time you open a terminal:
```

Author: I think there is also one innacuracy here... This could induce the user to repeat the process whenever it opens the terminal. 

I think it could be something like :

"If you wish the autocompletion to be enabled at the moment you start a terminal, use one of those methods: "

What do you think, @barriebyron ? :)

Reviewer: a ha! (where is the lightbulb emoji?) I see how it works now, I didn't review the commands careful. 

we can just tell the user to enable it and then give them the "enable now" and "persist" options. See how these suggested changes work.

Reviewer: I made the suggested change above ^^ line 1325

> 1. Source the completion script in your ~/.bashrc file:
> ```bash
> echo 'source <(starport  tools completions bash)' >> ~/.bashrc
> ```
> 
> 2. Add the completion script to the /etc/bash_completion.d directory:
> ```
> starport tools completions bash > /etc/bash_completion.d/starport
> ```

Reviewer: ```suggestion

- Source the completion script in your ~/.bashrc file:

    ```bash
    echo 'source <(starport  tools completions bash)' >> ~/.bashrc
    ```

- Add the completion script to the /etc/bash_completion.d directory:

    ```
    starport tools completions bash > /etc/bash_completion.d/starport
    ```

```

Reviewer: because these are alternate steps (one or the other) let's use unordered list items (bullets) instead of numbered steps

tip: to "nest" the command under the action step, you can indent. I used 4 spaces here, but in Markdown you can use the tab key

Reviewer: of course, now that we've provided clearer steps, we _could_ make these ordered steps again
